### DISCUSSION TITLE
[Question]: Is RequiredInputField being erroneously being called in these two situations?

### DISCUSSION THREAD

**Discussion by ronnyek** [ChilliCream/hotchocolate#5029] [General]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

`RequiredInputFieldIsMissing` seems to be getting fired in cases it seems like it shouldn't be. 
1. When a field is defined as say `public Optional<bool> IsAdmin {get;set;}` and object type doesn't explicitly override handling, I see this exception when omitting isAdmin from the mutation call. (if I define as Optional<bool?>, it doesn't throw that error, but imo seems like that shouldn't be required)
2. When an object type overrides handling of a property with code like `descriptor.Field(input => input.Class).Ignore();`, I get the same error saying required input field 'isAdmin' is missing.

### Steps to reproduce

1. Define some object and its object type
``` cs
// User.cs
public class User
{
   public string FirstName{get;set;}
   public string LastName{get;set;}
   public Optional<bool> IsAdmin{get;set;}
   public Optional<string> Class { get; set; }
}

//UserInputType.cs
public class UserInputType: InputObjectType<User> 
{
   protected override void Configure(IInputObjectTypeDescriptor<User> descriptor){
      descriptor.Field(input =>  input.Class).Ignore();
   }
}
```
2. build mutation type
``` cs 
// RootMutation.cs
public class RootMutationType: ObjectType<RootMutation>
{
   protected override void Configure(IObjectTypeDescriptor<RootMutation> descriptor)
   {
      descriptor.Field(i=>i.UpdateUser(default, default)).Argument("user", arg=>arg.Type<UserInputType>());
   }
}
```
3. Run query providing `FirstName`, `LastName`, `Class`, but omitting `IsAdmin` and expect error `required input field 'IsAdmin' is missing`
4. Run query providing `FirstName`, `LastName`, `IsAdmin` but omitting `Class` and expect error `required input field 'Class' is missing`

### Additional Context?

Its very possible this is a configuration issue, and I'm not sure if the descriptor for the mutation where default is passed into args of a method is correct (this is in updating code). I'm also nto even sure this stuff isnt the expected behavior, but I'd like to try understand what we'd expect to have happen in these situations.

Seems like this error comes from `CreateDefaultValue` in `InputParser.cs`
``` cs
 private object? CreateDefaultValue(InputField field, Path path, int stack)
    {
        if (field.DefaultValue is null || field.DefaultValue.Kind == SyntaxKind.NullValue)
        {
            if (field.Type.Kind == TypeKind.NonNull)
            {
                throw RequiredInputFieldIsMissing(field, path);
            }

            return field.IsOptional ? new Optional(null, false) : null;
        }
        // ....
 ```

To me it kind of seems like typekind is being set to `TypeKind.NonNull`, but for the purposes of this particular mutation (an update) I'd expect neither scenario to trigger this.

### Product

Hot Chocolate

### Version

12.7.0

**Comment by PascalSenn** [comment]
This works as designed.
Optional does not mean nullable :)
See https://github.com/ChilliCream/hotchocolate/issues/1808 for some more context

**Comment by ronnyek** [comment]
Optional doesnt mean nullable, but based on the behavior, it doesnt seem like it means optional either? Optional would imply that one could opt from providing it, and at least the entry point of the mtuation method would just not have that value be set

**Comment by PascalSenn** [comment]
No as a argument can be non nullable but still not have to be provided (DefaultValue)

**Comment by ronnyek** [comment]
Also, in the case where I'm explicitly telling hotchocolate to ignore what's being passed in for an update, is there a reason it should be generating a required field is missing? We ignore what they pass, but if they dont pass it, show validation errors?

**Comment by juan7732** [comment]
I've created https://github.com/ChilliCream/hotchocolate/pull/5013 in regards to what I saw in the slack channel from the discussion thread around this. Let me know if this helps.
